drafts/Asynchronous-programming-made-resilient-with-MPFT.adoc

```diff
@@ -0,0 +1,230 @@
+
+When you call the method, if there are less than the maximum concurrent executions running then your method is scheduled to run immediately, otherwise it's added to a queue. When one execution of the method finishes, if there are any on the queue then the first execution from the queue is started. If the queue itself is full, then the method fails with a `BulkheadException`.
+
+Just like the number of concurrent executions, the size of the queue can also be configured using the `waitingTaskQueue` parameter on the `@Bulkhead` annotation.
+
+== Deep dive: Asynchronous flow of execution
+This last section isn't critical for knowing how to use `@Asynchronous`, but it's useful stuff to know when working with MicroProfile Fault Tolerance. It covers how the flow of execution changes when using the `@Asynchronous` annotation compared to when it's not used. 
+
+Let's first look at how the Fault Tolerance annotations (`@Retry`, `@Timeout`, `@CircuitBreaker`, `@Bulkhead`,  and `@Fallback`) work together *without* the presence of `@Asynchronous` :
+
+image::/img/blog/FT-synchronous-execution-flow.png[Fault Tolerance synchronous execution flow]
+
+When the method is annotated with `@Asynchronous` a few things change. The differences from synchronous execution are highlighted in dark green, and discussed below:
+
+[caption="Accurate for mpFaultTolerance-2.0"]
+image::/img/blog/FT-asynchronous-execution-flow.png[Fault Tolerance asynchronous execution flow]
+
+The first difference compared to the synchronous flow of execution is that a `CompletionStage` or `Future` is returned before the method runs. When the method has actually returned, the result from the method is then propagated into the `CompletionStage` or `Future` so that the caller can get it. 
+
+The next difference comes in the Bulkhead. As well as either accepting or rejecting the execution, the Bulkhead can also queue it to be run later. If the method is accepted by the Bulkhead, it is then scheduled to be run on another thread, rather than run immediately. 
+
+Another difference occurs with Timeout. When used with @Asynchronous, if the Timeout expires, the method is interrupted, and the execution skips forwards to the point highlighted in the diagram (From the "Timeout Expires" block). The result is then processed as if the method had finished with a TimeoutException.
+
+The last difference is that if there's a Fallback, it also runs asynchronously, so it's scheduled to run on another thread as well.
+
+If a method returns a `CompletionStage` rather than a `Future`, then there's one more difference compared to the synchronous flow of execution. If the method returns a value rather than throwing an exception, we wait until that `CompletionStage` completes before doing the rest of the Fault Tolerance logic. A few things to point out here: 
```

Comment:
It is unclear what is the difference you tried to say. Do you mean if returning 'Future', we don't wait for Future to complete? It's not true though.